src/panels/calendar/ha-panel-calendar.js

```diff
@@ -0,0 +1,209 @@
+    ￼     height: 20px;
+    ￼     text-align: center;
+        }
+
+        span.calendar_color {
+          width: 15px;
+          height: 5px;
+     ￼    margin: 0px 10px 2px 0px;
+     ￼    display: inline-block;
+          background-color: #3174ad;
+        }
+     
+        span.calendar_color_spacer {
+          width: 15px;
+        }
+
+        .iron-selected {
+          background-color: #e5e5e5;
+          font-weight: normal;
+        }
+
+        paper-dialog > div.eventContent {
+          margin: 0px 5px 15px 5px;
+        }
+
+        paper-icon-item {
+          padding: 0px; 
+        }
+
+        paper-item-body.start {
+            padding: 0px 30px 0px 0px;
+        }
+
+        paper-dialog {
+            padding: 0px;
+        }
+
+        paper-dialog > h2 {
+            margin: 0;
+            padding: 20px;
+            background-color: #03A9F4;
+        }
+      </style>
+
+    <app-header-layout has-scrolling-region>
+      <app-header slot="header" fixed>
+        <app-toolbar>
+          <ha-menu-button narrow='[[narrow]]' show-menu='[[showMenu]]'></ha-menu-button>
+          <div main-title>[[localize('panel.calendar')]]</div>
+        </app-toolbar>
+      </app-header>
+
+      <div class="flex content">
+        <div id="calendars" class="layout vertical wrap">
+          <paper-card heading="Calendars">
+            <div class="all_calendars">
+              <paper-checkbox id="all_calendars" on-change="checkAll" checked>All calendars</paper-checkbox>
+            </div>
+            <paper-listbox id="calendar_list" multi on-selected-items-changed="_fetchData" selected-values="{{selectedCalendars}}" attr-for-selected="item-name">
+              <template is="dom-repeat" items="[[calendars]]">
+                <paper-item item-name="[[item.entity_id]]">
+                  <span class="calendar_color" style="background-color: [[item.color]]"></span>
+                  <span class="calendar_color_spacer"></span>
+                  [[item.name]]
+                </paper-item>
+              </template>
+            </paper-listbox>
+          </paper-card>
+        </div>
+        <div class="flex layout horizontal wrap">
+          <ha-big-calendar events="[[items]]"></ha-big-calendar>
+        </div>
+      </div>
+    </app-header-layout>
+    `;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._fetchData = this._fetchData.bind(this);
+    // TODO implement calendar_updated event
+    // this.hass.connection.subscribeEvents(this._fetchData, 'calendar_updated')
+    //  .then(function (unsub) { this._unsubEvents = unsub; }.bind(this));
+    this._fetchData();
+  }
+
+  _fetchData() {
+    // Fetch calendar list
+    this.hass.callApi('get', 'calendars')
+      .then((items) => {
+        this.calendars = items;
+      });
+    // Fetch events for selected calendar
+    const calls = this.selectedCalendars.map(cal => this.hass.callApi('get', 'calendar/' + cal));
```

Comment:
We should only fetch the events for the period of time that is currently being shown on the screen. Since there are recurring events, you can never fetch all events from a calendar.